### PR TITLE
Eslint fixing arrangements

### PR DESCRIPTION
**Pull request by rmoestl**
1) migrated to ESLint 4.8.0 (from old 2.10.1).
2) increased max line length to 120 (same as in another Plotly project I linted before) which made about 500 warnings go away.
3) ran `yarn install` to produce a new `yarn.lock` file.
4) updated `lint` scripts to also include the folders `backend`  and `scripts`

To test my changes, I ran the unit tests and started the desktop app locally successfully.

### CODE REVIEW

**Comment by bpostlethwaite** [comment]
@tarzzz we have some failing tests here. You know what they are about? https://circleci.com/gh/plotly/plotly-database-connector/2097

**Comment by tarzzz** [comment]
@bpostlethwaite from the failures, it seems the `before` hook of this test suites is failing, which is strange imho, because it only saves settings to local filesystem. Could there be some ESlint related issues here, which is causing failures @rmoestl ? .. :

```
    before(() => {
        saveSetting('USERS', [{username, apiKey}]);
        saveSetting('PLOTLY_API_DOMAIN', 'api.plot.ly');
        saveSetting('PLOTLY_API_SSL_ENABLED', true);
    });
```

**Comment by rmoestl** [comment]
@tarzzz tested that exhaustively on my machine. 
- First, ran unit tests with my commits included a couple of times. Result: 2 to 5 tests failed, arbitrarily it seemed. Just the MSSQL tests failing seems more common.
- Second, checkout master branch and ran unit tests again. Same result.
- Third, did a fresh clone of the repo. Ran a fresh `yarn install` and electron rebuild and ran units tests. Some tests also fail in this setting, e.g: 
  - MSSQL -  doesn't timeout if requestTimeout was set
  - mssql - tables - returns a list of tables

From what I'm seeing now, ESLint related commits don't have a negative effect on unit test. In fact, the commits don't touch actual source code by now.

**Comment by tarzzz** [comment]
Tests Look good to me.. except the merge conflicts:

image

**Comment by rmoestl** [comment]
We had a conflict in `yarn.lock` which is kind of hard to resolve by hand imho. I did an interactive rebase locally and build a new yarn.lock file in the same commit in which I changed ESLint dependencies. Ran unit tests and started the electron app once.

**Comment by rmoestl** [comment]
Conflicts resolved and tests pass. Ready to :dancer:, @tarzzz or @bpostlethwaite?

File: .eslintrc [hunk 1]
```diff
@@ -82,7 +82,9 @@
             "wrap-iife": [1],
             "yoda": [2],
             "key-spacing": [1],
-            "max-len": [1, 80, 4],
+            "max-len": ["error", 120, 4, {
```
Comment (tarzzz) [inline .eslintrc]: 👍

**Comment by bpostlethwaite** [comment]
:dancer:

**Pull request merged**